series/series.go

```diff
@@ -740,3 +746,186 @@ func (s Series) Quantile(p float64) float64 {
 
 	return stat.Quantile(p, stat.Empirical, ordered, nil)
 }
+
+// Histogram generates a mapping representing a histogram with bins
+// equally spaced binwidth in size.
+// If we did not specify a value for binwidth, i.e. 0 or negative
+// value passed, we compute binwidth ourselves to give a hopefully
+// sane number of bins.
+func (s Series) Histogram(binwidth float64) map[float64]int {
```

Comment:
I was actually thinking something along very similar lines. It felt like two series would make sense, but I did not see other examples of that being returned, probably because I did not look enough, and so I thought to go this route. It definitely seems like a two-series is a better answer, in my mind than a dataframe. I think for my purposes either a two-series or a map would be preferred over a dataframe, strictly based on how I can see that information being used. But, I also could make the argument that in a dataframe it becomes easier to persist this data for cases where one might just want to dump a two-column output, which I actually do a lot in C, and then use gnuplot to visualize.

Reply:
I thought about returning a DataFrame for Hists, and just don't see the value in doing so. I think two Series makes most sense to me honestly. I am going to play with this a bit today and see if maybe I am wrong about that. I also often refer to GSL as a source for inspiration for how to implement something along these lines and granted, in C we have much less flexibility in some ways, but they implement a struct with ranges array and bins array which maintains count. I think that sort of sounds like a DataFrame structure, when I rationalize your suggestion. I am still not sure about it, but I definitely think that Series is a step in the right direction over a map.